Fix some issues with file encoding while reading and writing the README.rst file when generating the html documentation
Index: pybind11-2.6.2/docs/conf.py
===================================================================
--- pybind11-2.6.2.orig/docs/conf.py
+++ pybind11-2.6.2/docs/conf.py
@@ -350,8 +350,7 @@ def generate_doxygen_xml(app):
 
 
 def prepare(app):
-    with open(DIR.parent / "README.rst") as f:
-        contents = f.read()
+    contents = (DIR.parent / "README.rst").read_text(encoding='utf-8')
 
     if app.builder.name == "latex":
         # Remove badges and stuff from start
@@ -360,8 +359,7 @@ def prepare(app):
         # Filter out section titles for index.rst for LaTeX
         contents = re.sub(r"^(.*)\n[-~]{3,}$", r"**\1**", contents, flags=re.MULTILINE)
 
-    with open(DIR / "readme.rst", "w") as f:
-        f.write(contents)
+    (DIR / "readme.rst").write_text(contents, encoding='utf-8')
 
 
 def clean_up(app, exception):
